Fix form theme template lookup support for style.html.twig

kuzmany opened the pull request:
## Summary
This PR adds support for the simplified `style.html.twig` template filename in form themes, following the same pattern as existing template lookups. This allows themes to use either the legacy `_style.html.twig` or the cleaner `style.html.twig` naming convention.

### Changes Made:
- Added template lookup for `style.html.twig` in FormModel.php
- Maintains backward compatibility with existing `_style.html.twig` templates
- Follows existing pattern used for other template lookups

## Test plan
- [ ] Create a custom theme with a `style.html.twig` file in `themes/[theme-name]/html/MauticFormBundle/Builder/`
- [ ] Configure a form to use this theme
- [ ] Verify that the style template is properly loaded and applied
- [ ] Test that existing themes with `_style.html.twig` continue to work

kuzmany committed:
fix: add style.html.twig template lookup support for form themes

Add support for style.html.twig template lookup in form themes,
following the same pattern as existing template lookups.
This allows themes to use the simplified style.html.twig filename
instead of the longer _style.html.twig convention.

escopecz commented on app/bundles/FormBundle/Model/FormModel.php:
```diff
@@ -423,6 +423,11 @@ public function generateHtml(Form $entity, $persist = true): string
+
+        if ($this->twig->getLoader()->exists('@themes/'.$theme.'/html/MauticFormBundle/Builder/style.html.twig')) {
+            $styleToRender = '@themes/'.$theme.'/html/MauticFormBundle/Builder/style.html.twig';
+        }
```
I know it's silly in this case since we are talking about 3 lines, but there is a lot of duplication. Basically just 1 character difference. Can we do something like this?
```suggestion
        foreach (['_style', 'style'] as $styleFile) {
            $stylePath = "@themes/{$theme}/html/MauticFormBundle/Builder/{$styleFile}.html.twig";
            if ($this->twig->getLoader()>exists($stylePath)) {
                $styleToRender = $stylePath;
            }
        }
```
Or use a private method.

kuzmany committed:
Merge branch '7.x' of https://github.com/mautic/mautic into private-pr-6624-fix-form-theme-style-template-support

kuzmany committed:
refactor: reduce duplication in style template lookup

Applied code review suggestion to use a foreach loop for checking
style template variations, reducing code duplication and improving
maintainability.
